compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -524,6 +524,7 @@ pub fn maybe_create_entry_wrapper<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
         // `main` should respect same config for frame pointer elimination as rest of code
         cx.set_frame_pointer_type(llfn);
         cx.apply_target_cpu_attr(llfn);
+        cx.apply_target_features(llfn);
```

Comment:
Maybe the `apply_target_*` calls and `set_frame_pointer_type` should be grouped together into a single function.

I noticed that `gen_fn` inside `compiler/rustc_codegen_llvm/src/intrinsic.rs` is also missing the `apply_target_features` by the way.